### PR TITLE
introduce `rustfmt`

### LINKED ISSUE #911: How about introduce `rustfmt` in github action?
This issue is not the feature.

## Motivation
Currently, we do not seem to have confirmed the format in github action.
(clippy lint  has already been introduced. https://github.com/SeaQL/sea-orm/pull/840)

## Proposed Solutions
Introducing CI

## Additional Information
 I would like to create a PR for this issue.

### PR DESCRIPTION
**Pull request by kyoto7250**
close #911 

## PR Info

This PR created a github aciton to test `rustfmt`.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -136,6 +136,26 @@ jobs:
...
+  rustfmt:
+    name: Rustfmt
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
```
Comment (kyoto7250) [inline .github/workflows/rust.yml:148]: `format_code_in_doc_comments` is the `nightly` channel only.

ref:
[format_code_in_doc_comments](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=format#format_code_in_doc_comments)

**Review by billy1624 — COMMENTED** [review]
Hey @kyoto7250, looks good! I think we can run cargofmt on all sea-orm crates and examples. Please check https://github.com/kyoto7250/sea-orm/compare/introduce_rustfmt...billy1624:sea-orm:pr/919

**Review by billy1624 — COMMENTED** [review]
I'm thinking can we create a shell script to run cargofmt on all sea-orm crates and examples? We could add that script to https://github.com/SeaQL/sea-orm/tree/master/build-tools

**Comment by kyoto7250** [comment]
@billy1624 

> I'm thinking can we create a shell script to run cargofmt on all sea-orm crates and examples? We could add that script to https://github.com/SeaQL/sea-orm/tree/master/build-tools

Sounds good, I will try it!

---

Currently, we use stable compiler in `Examples`, so the formatter fails in github action.

Should we switch toolchains to nightly version?

**Comment by billy1624** [comment]
> Currently, we use stable compiler in `Examples`, so the formatter fails in github action.
> 
> Should we switch toolchains to nightly version?

Can we install `stable` Rust side by side with `nightly` Rust? Then, we can run nightlyfmt with command `cargo +nightly fmt ...`

**Comment by baoyachi** [comment]
Oh，This MR is a bit contradictory to this sentence. https://github.com/SeaQL/sea-orm/issues/838#issuecomment-1174852956

**Comment by kyoto7250** [comment]
> Oh，This PR is a bit contradictory to this sentence. https://github.com/SeaQL/sea-orm/issues/838#issuecomment-1174852956

Thank you, I had not checked.

I think it is common to check the code style, but  I will abide by the maintainer's decision.

**Comment by kyoto7250** [comment]
Sorry, some jobs don't seem to stop, can you please force them to stop?

**Comment by kyoto7250** [comment]
Sorry, I didn't know it would work when I reopened the PR...🙇🏻

It seems to stop after about 20 minutes, so I'll run it just one more time for confirmation, then set a timeout...😥

**Comment by billy1624** [comment]
> Oh，This PR is a bit contradictory to this sentence. [#838 (comment)](https://github.com/SeaQL/sea-orm/issues/838#issuecomment-1174852956)

Hey @baoyachi, after second thoughts, I want to try introducing more cargo checks, e.g. cargofmt. We can always relex the CI checks if it's getting too noisy.

So, yeah, I changed my mind lolll

**Comment by baoyachi** [comment]
> > Oh，This PR is a bit contradictory to this sentence. [#838 (comment)](https://github.com/SeaQL/sea-orm/issues/838#issuecomment-1174852956)
> 
> Hey @baoyachi, after second thoughts, I want to try introducing more cargo checks, e.g. cargofmt. We can always relex the CI checks if it's getting too noisy.
> 
> So, yeah, I changed my mind lolll

So, as you said, why not add the cargo fix check together.

**Comment by kyoto7250** [comment]
> So, as you said, why not add the cargo fix check together.

It looks good idea for us.
How about we also add a script for running `clippy` in this PR?

**Comment by kyoto7250** [comment]
@billy1624 

I set a timeout and manually install nightly version.

Could you please check again?


> It looks good idea for us.
How about we also add a script for running clippy in this PR?

I tried my idea, but It is very slow (needs all project build), so I do not add.

**Review by billy1624 — APPROVED** [review]
Thanks! @kyoto7250 Perfect!

**Pull request merged**